### PR TITLE
simplify svc lister

### PR DESCRIPTION
**Pull request by deads2k**
trying to track down what killed the e2e tests.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33269)

### CODE REVIEW

**Comment by deads2k** [comment]
green e2e kicking again

**Comment by deads2k** [comment]
second green e2e, kicking again

**Comment by deads2k** [comment]
@k8s-bot test this: issue #IGNORE

**Comment by sttts** [comment]
@k8s-bot gce e2e test this

**Comment by sttts** [comment]
The last gce e2e problem was another flake I think.

File: pkg/controller/endpoint/endpoints_controller.go [hunk 1]
```diff
@@ -191,7 +191,7 @@ func (e *EndpointController) getPodServiceMemberships(pod *api.Pod) (sets.String
 		return set, nil
 	}
 	for i := range services {
-		key, err := keyFunc(&services[i])
+		key, err := keyFunc(services[i])
```
Comment (smarterclayton) [inline pkg/controller/endpoint/endpoints_controller.go]: Clutch

**Comment by smarterclayton** [comment]
Changes LGTM

**Pull request merged**